CDTDatastoreReplicationAcceptanceTests/ReplicationAcceptance.m

```diff
@@ -1104,6 +1104,40 @@ -(void) testPullDeleteAllPush
 }
 
 /**
+ * Test ensures deleted documents correctly pulled
+ * - Create local documents
+ * - Push to remote
+ * - Delete some
+ * - Pull replicate back
+ * - Compare DBs
+ */
+-(void) testPushDeleteSomePull
+{
+    // Create docs in remote database
+    NSLog(@"Creating local documents...");
+    [self createLocalDocs:self.n_docs];
+    [self pushToRemote];
+    XCTAssertEqual(self.n_docs, self.datastore.documentCount, @"Incorrect number of documents created");
+    
+    // delete half the remote docs
+    // (remote deletes are really slow so we only do a small proportion)
+    NSLog(@"Deleting remote docs...");
+    for (int i = 1; i < self.n_docs/10; i++) {
+        NSString *docId = [NSString stringWithFormat:@"doc-%i", i];
+        [self deleteRemoteDocWithId:docId];
+    }
+    
+    NSLog(@"Pull replicate...");
+    // Replicate the changes
+    [self pullFromRemote];
+    
+    NSLog(@"Compare...");
+    BOOL same = [self compareDatastore:self.datastore
```

Comment:
I think it would add a little value to also do this comparison before the `pullFromRemote`, and then assert that they're different, although I guess we have other tests that test that delete and pull do the right things, so it's not essential.